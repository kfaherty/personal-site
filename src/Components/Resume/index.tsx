import * as React from 'react';
import { css, default as styled } from 'styled-components';

import { media } from '../GlobalStyle/media';

interface IPositionedCell {
  position?: number;
  span?: number;
}

const ResumeWrap = styled.div`
  margin: 3em;
`;

const Grid = styled.div`
  display: grid;
  gap: 1em 1em;
  grid-template-columns: 100%;
  margin-bottom: 3em;

  ${media.small`
    grid-template-columns: 10em 25%;
  `}
  ${media.medium`
    grid-template-columns: 10em repeat(2, 25%);
  `}
  ${media.large`
    grid-template-columns: 10em repeat(3, 25%);
  `}
`;

const HeaderGrid = styled(Grid)`
  align-items: center;
  margin: 5em 0;
`;

// tslint:disable: no-magic-numbers
const PositionedCell = styled.div<IPositionedCell>`
  grid-column-start: ${({ position }) => position ? position : 'auto'};
  grid-column-end: span 1;

  ${({ span }) => span && css`
    ${media.small`
      grid-column-end: ${span > 2 ? 'span 2' : `span ${span}`};
    `}
    ${media.medium`
      grid-column-end: ${span > 3 ? 'span 3' : `span ${span}`};
    `}
    ${media.large`
      grid-column-end: ${span > 4 ? 'span 4' : `span ${span}`};
    `}
  `}
`;
// tslint:enable: no-magic-numbers

const Headline = styled.h1`
  font-size: 4.2rem;
  line-height: 3.6rem;
  margin: 0;
`;

const Subhead = styled.h2``;

const Link = styled.a`
  font-style: italic;
  color: ${({ theme: { link }}) => link.colorLink};

  &:hover {
    color: ${({ theme: { link }}) => link.colorLinkHover};
  }
`;

const Section = styled.h3`
  display: inline-flex;
  width: min-content;
  white-space: nowrap;
  padding: 0.125em 0.375em;
  background: ${({ theme: { section } }) => section.colorBackground};
  color: ${({ theme: { section } }) => section.colorText};
  align-self: baseline;
  justify-self: flex-end;
  text-transform: uppercase;
  font-size: 1.6rem;
  line-height: 2.2rem;
  margin-top: -0.125em;
`;

const CellHeader = styled.h4`
  font-weight: 700;
  font-size: 1.6rem;
  line-height: 2.2rem;
`;

const ListBullet = styled.ul`
  margin: 1em 1.25em;
  list-style: none;
  color: ${({ theme: { text } }) => text.colorText};
`;

const BulletElement = styled.li`
  padding-left: 0.25em;
  margin: 0 0 0.5em;
  position: relative;

  &::before {
    content: '\\2022';
    position: absolute;
    color: ${({ theme }) => theme.text.colorText};
    left: -0.75em;
    font-size: 2.5rem;
  }
`;

const Resume: React.FC = () => (
  <ResumeWrap>
    <HeaderGrid>
      <div>
        logo
      </div>
      <PositionedCell span={2}>
        <Headline>Kevin Faherty</Headline>
        <Subhead>Full-stack Javascript Engineer</Subhead>
      </PositionedCell>
      <div>
        <div>hello@kevinfaherty.com</div>
        <Link href='https://github.com/kfaherty'>https://github.com/kfaherty</Link>
        <div>Chicago, IL</div>
      </div>
    </HeaderGrid>
    <Grid>
      <Section>Work Experience</Section>
      <div>
        <CellHeader>Software Engineering Lead</CellHeader>
        <div>Amount - <Link href='amount.com'>amount.com</Link></div>
        <div>Chicago, IL</div>
        <div>October 2019 - Present</div>
      </div>
      <div>
        <CellHeader>Software Engineer</CellHeader>
        <div>Avant/Amount - <Link href='avant.com'>avant.com</Link></div>
        <div>Chicago, IL</div>
        <div>February 2018 - October 2019</div>
      </div>
      <div>
        <CellHeader>Lead Javascript Developer</CellHeader>
        <div>Riskbone - <Link href='leveltradingfield.com'>leveltradingfield.com</Link></div>
        <div>Chicago, IL</div>
        <div>March 2016 - October 2017</div>
      </div>
      <PositionedCell position={2}>
        <CellHeader>Web Developer / Designer</CellHeader>
        <div>Freelance</div>
        <div>Milwaukee, WI</div>
        <div>October 2015 - April 2016</div>
      </PositionedCell>
      <div>
        <CellHeader>Graphic Designer</CellHeader>
        <div>Radyne Corporation</div>
        <div>Milwaukee, WI</div>
        <div>February 2014 - October 2015</div>
      </div>
    </Grid>
    <Grid>
      <Section>Skills</Section>
      <div>
        <div>Typescript / ESNext / React</div>
        <div>WebSockets / GraphQL / REST</div>
        <div>Node.JS / Express / PostgreSQL</div>
      </div>
      <div>
        <div>PostCSS / SCSS / HTML5 / WCAG</div>
        <div>Webpack / AWS / Jenkins / Grunt</div>
        <div>Visual Regression / Snapshot Testing</div>
      </div>
      <div>
        <div>Angular / Backbone / jQuery</div>
        <div>Photoshop / Illustrator / InDesign</div>
        <div>AfterEffects / Maya / Unity</div>
      </div>
    </Grid>
    <Grid>
      <Section>Education</Section>
      <PositionedCell span={2}>
        <CellHeader>Milwaukee Institute of Art and Design</CellHeader>
        <div>Bachelor of Fine Arts - Emphasis in Illustration</div>
        <div>Milwaukee, WI</div>
        <div>Graduated December 2013</div>
      </PositionedCell>
    </Grid>
    <Grid>
      <Section>References</Section>
      <p>Available on request</p>
    </Grid>
  </ResumeWrap>
);

export default Resume;
