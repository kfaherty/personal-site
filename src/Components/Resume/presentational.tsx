import { css, default as styled } from 'styled-components';

import { media } from '../GlobalStyle/media';

interface IPositionedCell {
  position?: number;
  span?: number;
}

export const ResumeWrap = styled.div`
  margin: 5em 3em;
  display: grid;
  gap: 5em 1em;
  grid-template-columns: 100%;
  grid-template-rows: 8em auto;

  ${media.small`
    grid-template-columns: 10em calc(100% - 10em);
  `}
`;

export const Logo = styled.img`
  align-items: center;
  height: 8em;
`;

export const Grid = styled.div`
  display: grid;
  gap: 1em 1em;
  grid-template-columns: 100%;

  ${media.small`
    grid-template-columns: 100%;
  `}
  ${media.medium`
    grid-template-columns: repeat(2, 50%);
  `}
  ${media.large`
    grid-template-columns: repeat(3, 33%);
  `}
`;

export const HeaderGrid = styled(Grid)`
  align-items: center;
`;

// tslint:disable: no-magic-numbers
export const PositionedCell = styled.div<IPositionedCell>`
  grid-column-start: ${({ position }) => position ? position : 'auto'};
  grid-column-end: span 1;

  ${({ span }) => span && css`
    ${media.small`
      grid-column-end: ${span > 1 ? 'span 2' : `span ${span}`};
    `}
    ${media.medium`
      grid-column-end: ${span > 2 ? 'span 3' : `span ${span}`};
    `}
    ${media.large`
      grid-column-end: ${span > 3 ? 'span 4' : `span ${span}`};
    `}
  `}
`;
// tslint:enable: no-magic-numbers

export const Headline = styled.h1`
  font-size: 4.2rem;
  line-height: 3.6rem;
  margin: 0;
`;

export const Subhead = styled.h2``;

export const Link = styled.a`
  font-style: italic;
  color: ${({ theme: { link }}) => link.colorLink};

  &:hover {
    color: ${({ theme: { link }}) => link.colorLinkHover};
  }
`;

export const Section = styled.h3`
  display: inline-flex;
  width: min-content;
  white-space: nowrap;
  padding: 0.125em 0.375em;
  background: ${({ theme: { section } }) => section.colorBackground};
  color: ${({ theme: { section } }) => section.colorText};
  align-self: baseline;
  text-transform: uppercase;
  font-size: 1.6rem;
  line-height: 2.2rem;
  margin-top: -0.125em;

  ${media.small`
    justify-self: flex-end;
  `}
`;

export const CellHeader = styled.h4`
  font-weight: 700;
  font-size: 1.6rem;
  line-height: 2.2rem;
`;

export const List = styled.ul`
  list-style: none;
  color: ${({ theme: { text } }) => text.colorText};
`;

export const ListBullet = styled(List)`
  margin: 1em 1.25em;
`;

export const ListElement = styled.li`
  margin: 0 0 0.5em;
  position: relative;
`;

export const BulletElement = styled(ListElement)`
  padding-left: 0.25em;

  &::before {
    content: '\\2022';
    position: absolute;
    color: ${({ theme }) => theme.text.colorText};
    left: -0.75em;
    font-size: 2.5rem;
  }
`;
