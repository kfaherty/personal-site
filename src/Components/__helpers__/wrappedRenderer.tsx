
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import { ThemeProvider } from 'styled-components';

import { theme } from '../GlobalStyle/theme';

export const wrappedRenderer: (
  nextElement: React.ReactElement,
  options?: { location?: string }
) => renderer.ReactTestRenderer = (nextElement, options) => renderer.create(
  (
    <ThemeProvider theme={theme}>
      <MemoryRouter initialEntries={[(options && options.location) || '/']}>
        {nextElement}
      </MemoryRouter>
    </ThemeProvider>
  )
);
