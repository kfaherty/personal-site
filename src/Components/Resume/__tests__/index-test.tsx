import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { wrappedRenderer } from '../../__helpers__/wrappedRenderer';
import Resume from '../';

describe('Resume Component', () => {
  it('matches snapshot', () => {
    const component: renderer.ReactTestRenderer = wrappedRenderer(
      <Resume />
    );
    expect(component.toJSON())
      .toMatchSnapshot();
  });
});
