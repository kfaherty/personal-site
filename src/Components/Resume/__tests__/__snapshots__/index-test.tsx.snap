// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume Component matches snapshot 1`] = `
<div
  className="c0"
>
  <img
    alt="logo"
    className="c1"
    src="logo.jpg"
  />
  <div
    className="c2"
  >
    <div
      className="c3"
      span={2}
    >
      <h1
        className="c4"
      >
        Kevin Faherty
      </h1>
      <h2
        className=""
      >
        Full-stack Javascript Engineer
      </h2>
    </div>
    <div>
      <div>
        hello@kevinfaherty.com
      </div>
      <a
        className="c5"
        href="https://github.com/kfaherty"
      >
        https://github.com/kfaherty
      </a>
      <div>
        Chicago, IL
      </div>
    </div>
  </div>
  <h3
    className="c6"
  >
    Work Experience
  </h3>
  <div
    className="c7"
  >
    <div>
      <h4
        className="c8"
      >
        Software Engineering Lead
      </h4>
      <div>
        Amount - 
        <a
          className="c5"
          href="https://amount.com"
        >
          amount.com
        </a>
      </div>
      <div>
        Chicago, IL
      </div>
      <div>
        October 2019 - Present
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Lead developer on CRM and partner-facing applications
        </li>
        <li
          className="c10"
        >
          Introduced API integrations with S3, Okta, Jira, and Ruby on Rails
        </li>
        <li
          className="c10"
        >
          Developed front-end of point of sale and in-branch loan products using AWS Lambda and Serverless
        </li>
        <li
          className="c10"
        >
          Created multi-tenant module framework for scaling Partner Portal
        </li>
        <li
          className="c10"
        >
          Executed Amount brand redesign of partner-facing applications
        </li>
        <li
          className="c10"
        >
          Performed extensive code review and mentoring of junior developers
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Software Engineer
      </h4>
      <div>
        Avant/Amount - 
        <a
          className="c5"
          href="https://avant.com"
        >
          avant.com
        </a>
      </div>
      <div>
        Chicago, IL
      </div>
      <div>
        February 2018 - October 2019
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Front-end engineer responsible for crucial contributions to Avant/Amount’s customer and partner facing front-end applications built with Typescript, React, Redux, Node.JS, and GraphQL.
        </li>
        <li
          className="c10"
        >
          Overhauled styling pipeline to a themable component library with React Styled-Components
        </li>
        <li
          className="c10"
        >
          Modernized CRM product to use Apollo GraphQL and Jest
        </li>
        <li
          className="c10"
        >
          Developed customer dashboard for self-servicing of loan and credit card
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Lead Javascript Developer
      </h4>
      <div>
        Riskbone - 
        <a
          className="c5"
          href="https://leveltradingfield.com"
        >
          leveltradingfield.com
        </a>
      </div>
      <div>
        Chicago, IL
      </div>
      <div>
        March 2016 - October 2017
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Front-end lead responsible for creating an online trading platform with Backbone, Socket.io, SCSS, jQuery UI, ZeroMQ, and Node.JS
        </li>
        <li
          className="c10"
        >
          Managed four team members to meet development deadlines
        </li>
        <li
          className="c10"
        >
          Built key application frameworks for authentication, payment processing, caching, pub-sub, and messaging
        </li>
        <li
          className="c10"
        >
          Created desktop UI with workspaces
        </li>
        <li
          className="c10"
        >
          Integrated with C++ API to render real-time market data, insert orders, and calculate PnL efficiently
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Web Developer / Designer
      </h4>
      <div>
        Freelance
      </div>
      <div>
        Milwaukee, WI
      </div>
      <div>
        October 2015 - April 2016
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Freelanced on web projects requiring web/mobile design, responsive CSS layouts, seamless page transitions, templating, e- commerce, asynchronous loading, prototyping, SEO, content management systems, and more
        </li>
        <li
          className="c10"
        >
          Commissioned as a designer to create illustrations, advertisements, brochures, posters, brand identities, icons, logos, and more
        </li>
        <li
          className="c10"
        >
          Participated in the local community of freelance professionals
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Graphic Designer
      </h4>
      <div>
        Radyne Corporation
      </div>
      <div>
        Milwaukee, WI
      </div>
      <div>
        February 2014 - October 2015
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Independently designed marketing content in-house for print and web
        </li>
        <li
          className="c10"
        >
          Responsible for developing websites, product icons, animations, UI wireframes, product manuals/guides, brochures and catalogs, advertisements, video and photography content, email templates, and exhibition designs
        </li>
        <li
          className="c10"
        >
          Modernized the parent company’s branding guidelines
        </li>
        <li
          className="c10"
        >
          Created a distinct graphic identity for Radyne within brand standards
        </li>
      </ul>
    </div>
  </div>
  <h3
    className="c6"
  >
    Projects
  </h3>
  <div
    className="c7"
  >
    <div>
      <h4
        className="c8"
      >
        ArtHawk
      </h4>
      <div>
        Artist Showcase Platform - 
        <a
          className="c5"
          href="https://art-hawk.com"
        >
          Website
        </a>
      </div>
      <div>
        2020
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          An art showcase platform with built-in tools for freelance artists
        </li>
        <li
          className="c10"
        >
          Features a global gallery for discovering new artists, comment threads, favorites, and a commission workflow for hiring freelance artists
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Squawk Chat
      </h4>
      <div>
        Chat Client - 
        <a
          className="c5"
          href="https://github.com/kfaherty/squawk-chat"
        >
          Github
        </a>
      </div>
      <div>
        2018
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          A chat client written in React that integrates with a third-party websocket API
        </li>
        <li
          className="c10"
        >
          Features realtime chat messaging in private and public rooms, direct messages, animations, and more
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Supervisitor
      </h4>
      <div>
        Jam Game - 
        <a
          className="c5"
          href="https://mattdonatelli.itch.io/supervisitor"
        >
          Website
        </a>
      </div>
      <div>
        2017
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Supervisitor is a game made in 48 hours for the 
          <a
            className="c5"
            href="https://itch.io/jam/bscotch100"
          >
            Butterscotch ShenaniJam
          </a>
           made in Unity using C# with Matt Donatelli and Zach Fendelman
        </li>
        <li
          className="c10"
        >
          Contributed environment art, level design, lighting, and modeling
        </li>
        <li
          className="c10"
        >
          Ranked #6 overall by user ratings out of 130 entries
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        TweetBox
      </h4>
      <div>
        Twitter Client - 
        <a
          className="c5"
          href="https://github.com/kfaherty/tweets-react"
        >
          Github
        </a>
      </div>
      <div>
        2017
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          Twitter app that features oAuth authentication with Twitter, realtime streaming timelines, favorites, retweets, mentions, direct messages, profiles, search, and more
        </li>
        <li
          className="c10"
        >
          Built using React, socket.io, and a Node backend
        </li>
      </ul>
    </div>
    <div>
      <h4
        className="c8"
      >
        Our Forgotten Alphabet
      </h4>
      <div>
        Arcade Game - 
        <a
          className="c5"
          href="https://github.com/kfaherty/ourforgottenalphabet"
        >
          Github
        </a>
      </div>
      <div>
        2015
      </div>
      <ul
        className="c9"
      >
        <li
          className="c10"
        >
          A game about a comet drawing constellations, made in the Love2d framework using Lua
        </li>
        <li
          className="c10"
        >
          Based on an unreleased game by Steph Thirion
        </li>
      </ul>
    </div>
  </div>
  <h3
    className="c6"
  >
    Skills
  </h3>
  <ul
    className="c11"
  >
    <li
      className="c12"
    >
      Typescript / ESNext / React
    </li>
    <li
      className="c12"
    >
      WebSockets / GraphQL / REST
    </li>
    <li
      className="c12"
    >
      Node.JS / Express / PostgreSQL
    </li>
    <li
      className="c12"
    >
      PostCSS / SCSS / HTML5 / WCAG
    </li>
    <li
      className="c12"
    >
      Webpack / AWS / Jenkins / Grunt
    </li>
    <li
      className="c12"
    >
      Visual Regression / Snapshot Testing
    </li>
    <li
      className="c12"
    >
      Angular / Backbone / jQuery
    </li>
    <li
      className="c12"
    >
      Photoshop / Illustrator / InDesign
    </li>
    <li
      className="c12"
    >
      AfterEffects / Maya / Unity
    </li>
  </ul>
</div>
`;
